FontEditor: Support flipping and rotating, fix pasting, and other cosmetic errors

BenWiederhake opened the pull request:
Follow-up to #10964

![Bildschirmfoto_2021-11-20_03-11-16](https://user-images.githubusercontent.com/2690845/142710853-53d9025d-292f-47a5-a24c-08b36275ab46.png)

awesomekling commented:
Code changes seem all right, but the commit messages are a bit weak.

"FontEditor: Make fixed height implicit" sounds like it might have something to do with font/glyph heights. Instead it's a UI layout tweak.

"FontEditor: Acknowledge that undo state always exists" sounds like it should always be possible to undo. Instead it's about `save_state()` always succeeding.

"FontEditor: Fix glyph pasting" is missing an explanation of the bug being fixed.

BenWiederhake commented:
Changes since last push:
- Better commit descriptions (thank @awesomekling!)
- Rebased for convenience

The CI failure in GA x86_64 seems unrelated: This PR doesn't touch test-js or the Kernel, and the crash seems to be due to concurrent changes to Kernel Regions, in particular:
```
1014.926 [#0 test-js(503:503)]: kmalloc: Adding even more memory to heap at V0x0000002006dd9000, bytes: 2625536
1014.986 [#0 test-js(503:503)]: Unrecoverable page fault, write to address V0x00000014a5e6f000
1014.986 [test-js(503:503)]: CRASH: CPU #0 Page Fault in ring 3
1014.986 [#0 test-js(503:503)]: Exception code: 0007 (isr: 0000)
1014.986 [#0 test-js(503:503)]:     pc=0x0013:0x00000009c1b29e41 rflags=0x0000000000000282
1014.986 [#0 test-js(503:503)]:  stack=0x00000000011fec08
1014.986 [#0 test-js(503:503)]:    rax=0xdcdcdcdcdcdcdcdc rbx=0x0000000000000001 rcx=0x0000000000000004 rdx=0x0000000000000020
1014.986 [#0 test-js(503:503)]:    rbp=0x00000000011fed50 rsp=0x0000002003612ec0 rsi=0x00000000000000dc rdi=0x00000014a5e6f000
1014.986 [#0 test-js(503:503)]:     r8=0x00000014a5e6f000  r9=0x0000000000dc0000 r10=0x0000000000000001 r11=0x00000009c2091540
1014.986 [#0 test-js(503:503)]:    r12=0x00000009c21dfed8 r13=0x0000000000000020 r14=0x00000014a5e60000 r15=0x000000000000077f
1014.986 [#0 test-js(503:503)]:    cr0=0x0000000080010013 cr2=0x00000014a5e6f000 cr3=0x0000000010ebd000 cr4=0x00000000003406e4
… SNIP …
1014.986 [#0 test-js(503:503)]: 0x00000014a2e30000 -- 0x00000014a2e33fff 0x0000000000004000 RW     LibJS: HeapBlock(256)
1014.986 [#0 test-js(503:503)]: 0x0000000000000000 -- 0xffffffffffffffff 0x0000000000000000        
1014.986 [#0 test-js(503:503)]: 0x00000014a8be0000 -- 0x00000014a8beffff 0x0000000000010000 RW     malloc: ChunkedBlock(2032)
```
